src/modules/navigator/takeoff.cpp

```diff
@@ -104,10 +104,17 @@ Takeoff::set_takeoff_position()
 
 	float abs_altitude = 0.0f;
 
-	const float min_abs_altitude = _navigator->get_home_position()->alt + _param_min_alt.get();
+	static float min_abs_altitude;
 
-	// Use altitude if it has been set.
-	if (rep->current.valid && PX4_ISFINITE(rep->current.alt)) {
+	if (_navigator->home_position_valid()) { //only use home position if it is valid
+		min_abs_altitude = _navigator->get_home_position()->alt + _param_min_alt.get();
+
+	} else { //e.g. flow
+		min_abs_altitude = _param_min_alt.get();
+	}
+
+	// Use altitude if it has been set. If home position is invalid use min_abs_altitude
+	if (rep->current.valid && PX4_ISFINITE(rep->current.alt) && _navigator->home_position_valid()) {
```

Comment:
I think you can still leave the if statement with home_position_valid which tells us that we have gps. But I would substitute get_home_position with get_global_position since the takeoff should be relative to the current altitude and not the home altitude